lib/domain.js

```diff
@@ -59,6 +59,20 @@ Domain.prototype._disposed = undefined;
 // Called by process._fatalException in case an error was thrown.
 Domain.prototype._errorHandler = function errorHandler(er) {
   var caught = false;
+  var self = this;
+
+  function emitError() {
+    var handled = self.emit('error', er);
```

Comment:
It would be great to automate this rule with our linting tools. Is this something that has been investigated before?

Using the [prefer-const])(http://eslint.org/docs/rules/prefer-const) rule [does not apply to function-scoped variable declarations](https://github.com/eslint/eslint/blob/master/lib/rules/prefer-const.js#L56), so we would need to use [the no-var eslint rule](http://eslint.org/docs/rules/no-var) too, which I assume would require a significant amount of work. I'm also not sure about all the implications since I'm not familiar with ES6.

I'm not a big fan of having implicit coding style rules, because it becomes frustrating for maintainers to enforce them every time, and for contributors because they don't have any tool to check that their code complies to the guidelines.

@thefourtheye  @nodejs/tsc Any thoughts on this?